contracts/settings/ATMSettings.sol

```diff
@@ -0,0 +1,243 @@
+            atmGovernanceFactoryAddress != address(0x0),
+            "ATM_GOV_FACTORY_MUST_BE_PROVIDED"
+        );
+        require(settingsAddress != address(0x0), "SETTINGS_MUST_BE_PROVIDED");
+
+        atmGovernanceFactory = ATMGovernanceFactoryInterface(atmGovernanceFactoryAddress);
+        settings = SettingsInterface(settingsAddress);
+    }
+
+    /** External Functions */
+
+    /**
+        @notice It pauses an given ATM.
+        @param atmAddress ATM address to pause.
+     */
+    function pauseATM(address atmAddress)
+        external
+        withPauserRole()
+        withValidATM(atmAddress)
+    {
+        require(settings.isPaused() == false, "PLATFORM_IS_ALREADY_PAUSED");
+        require(atmToPaused[atmAddress] == false, "ATM_IS_ALREADY_PAUSED");
+
+        atmToPaused[atmAddress] = true;
+
+        emit ATMPaused(atmAddress, msg.sender);
+    }
+
+    /**
+        @notice It unpauses an given ATM.
+        @param atmAddress ATM address to unpause.
+     */
+    function unpauseATM(address atmAddress)
+        external
+        withPauserRole()
+        withValidATM(atmAddress)
+    {
+        require(settings.isPaused() == false, "PLATFORM_IS_PAUSED");
+        require(atmToPaused[atmAddress] == true, "ATM_IS_NOT_PAUSED");
+
+        atmToPaused[atmAddress] = false;
+
+        emit ATMUnpaused(atmAddress, msg.sender);
+    }
+
+    /**
+        @notice Gets whether an ATM is paused (or the platform is paused) or not.
+        @param atmAddress ATM address to test.
+        @return true if ATM is paused. Otherwise it returns false.
+     */
+    function isATMPaused(address atmAddress) external view returns (bool) {
+        return settings.isPaused() || atmToPaused[atmAddress];
+    }
+
+    /**
+        @notice Sets an ATM for a given market (borrowed token and collateral token).
+        @param borrowedToken borrowed token address.
+        @param collateralToken collateral token address.
+        @param atmAddress ATM address to set.
+     */
+    function setATMToMarket(
+        address borrowedToken,
+        address collateralToken,
+        address atmAddress
+    ) external withPauserRole() withValidATM(atmAddress) {
+        require(borrowedToken.isContract() == true, "BORROWED_TOKEN_MUST_BE_CONTRACT");
+        require(collateralToken.isContract() == true, "COLL_TOKEN_MUST_BE_CONTRACT");
+        require(
+            marketToAtm[borrowedToken][collateralToken] == address(0x0),
+            "ATM_TO_MARKET_ALREADY_EXIST"
+        );
+
+        marketToAtm[borrowedToken][collateralToken] = atmAddress;
+
+        emit MarketToAtmSet(borrowedToken, collateralToken, atmAddress, msg.sender);
+    }
+
+    /**
+        @notice Updates a new ATM for a given market (borrowed token and collateral token).
+        @param borrowedToken borrowed token address.
+        @param collateralToken collateral token address.
+        @param newAtmAddress the new ATM address to update.
+     */
+    function updateATMToMarket(
```

Comment:
Given we are going to make the ATMs upgradeable, do we want to be able to change the ATM proxy address still?

Reply:
In this case yes, because we might use a different ATM (insurance atm instead of real state) for a given market.